client-java/sql/src/main/java/org/evomaster/client/java/sql/internal/SqlParserUtils.java

```diff
@@ -0,0 +1,95 @@
+package org.evomaster.client.java.sql.internal;
+
+import net.sf.jsqlparser.JSQLParserException;
+import net.sf.jsqlparser.expression.Expression;
+import net.sf.jsqlparser.parser.CCJSqlParserUtil;
+import net.sf.jsqlparser.statement.Statement;
+import net.sf.jsqlparser.statement.delete.Delete;
+import net.sf.jsqlparser.statement.select.PlainSelect;
+import net.sf.jsqlparser.statement.select.Select;
+import net.sf.jsqlparser.statement.update.Update;
+
+public class SqlParserUtils {
+
+    /**
+     * We only use the selects that refer to objects in the database that are meaningful for testing purposes,
+     * when code access to a sequence for example when getting the next id for a new object in the table,
+     * then we don't want to use that select as a target.
+     * @param sqlCommand
+     * @return
+     */
+    public static boolean isSelect(String sqlCommand) {
+        return startsWithIgnoreCase(sqlCommand, "select") && !isASequence(sqlCommand);
+    }
+
+    public static boolean isDelete(String sqlCommand) {
+        return startsWithIgnoreCase(sqlCommand, "delete");
+    }
+
+    public static boolean isUpdate(String sqlCommand) {
+        return startsWithIgnoreCase(sqlCommand, "update");
+    }
+
+    public static boolean isInsert(String sqlCommand) {
+        return startsWithIgnoreCase(sqlCommand, "insert");
+    }
+
+    private static boolean startsWithIgnoreCase(String input, String prefix){
+        return input!= null && input.trim().toLowerCase().startsWith(prefix);
+    }
+
+    private static boolean isASequence(String input) {
+        return input!= null && input.trim().toLowerCase().matches(".*(currval|nextval).*");
+    }
+
+    /**
+     * check if the sql is `Select 1`
+     * detected by proxyprint as
+     *      ERROR - FAILED TO COMPUTE HEURISTICS FOR SQL: SELECT 1
+     *
+     * https://stackoverflow.com/questions/3668506/efficient-sql-test-query-or-validation-query-that-will-work-across-all-or-most
+     */
+    public static boolean isSelectOne(String sqlCommand) {
+        return sqlCommand!= null && sqlCommand.trim().toLowerCase().matches("select\\s+1\\s*;?");
```

Comment:
wouldn't this work for any number, and not just `1`? ie, have a more generic `select\\s+-?\\d+\\s*;?`

Reply:
makes sense